### DISCUSSION TITLE
Bank Transfer Gateway Support

### DISCUSSION THREAD

**Discussion by benjibee** [duncanmcclean/simple-commerce#760] [Help] [answered]
## Description
I'm looking to accept payments via bank transfer. This means an order isn't marked as paid after checkout, and an additional notification would need to be send out when an order is placed. My current solution is incomplete, and I'm thinking it might be worthwhile adding this as a default payment gateway included with Simple Commerce.

## Issues
- Orders not marked as paid are seen as orders that weren't made, same as abandoned carts.
- There's no built-in configuration option for an `order_made` event to send out a notification.

## Attempts
I've added a custom gateway and made an attempt at adding a listener to the `post_checkout` event but we also accept credit card payments and sending out a notification in same manner as the supported events has got me a bit lost.

I'm going to continue building this custom workaround but I really think this could help more than a few users out there. I'm curious to get feedback about this approach.

**Comment by duncanmcclean** [comment, marked as answer]
Thanks for creating this discussion!

I'll see what I can do regarding creating an order status between an order being placed and an order being paid. As you've noticed, they're the same thing right now. I probably won't be able to introduce it until v5 as it'll be a (small) breaking change.

I'll have a think about a Bank Transfer gateway as well - although, it should be easy to do manually once the 'in-between' status is in place.

I'll update this discussion when either/both of the 2 have been released!

**Reply by benjibee** [reply]
That's great, thank you! I'm sure having this status and default gateway will help others who handle payment offsite (not just with redirects) as most of Germany still uses bank transfer for these sorts of payments.

Could you recommend a good approach to sending out a notification email when an order is made with a specific gateway? As mentioned I tried the listener, I also read through your PR adding additional events for notifications, but don't quite have a solid solution.

**Reply by duncanmcclean** [reply]
I'd probably listen to the `OrderPaid` event, then check the `->gateway()` on the `Order` you get from the event. If it's the one you're after, then do your logic, if it's not, return early.

**Reply by benjibee** [reply]
The issue is that I would need to send out the notification for a bank transfer before the order is marked as paid. But I guess I could also mark it as paid, and then if it matches the conditional unmark it?

**Reply by duncanmcclean** [reply]
Yeah, you could do. 

There's also the `PostGateway` event you could listen for to do the same thing. Although, you'd have to check if it works for your case.
